test/functional/lcow_container_bench_test.go

```diff
@@ -16,32 +17,61 @@ import (
 	"github.com/Microsoft/hcsshim/internal/hcsoci"
 	"github.com/Microsoft/hcsshim/internal/layers"
 	"github.com/Microsoft/hcsshim/internal/resources"
+	"github.com/Microsoft/hcsshim/internal/uvm"
 	"github.com/Microsoft/hcsshim/osversion"
 
 	testcmd "github.com/Microsoft/hcsshim/test/internal/cmd"
 	"github.com/Microsoft/hcsshim/test/internal/container"
 	testlayers "github.com/Microsoft/hcsshim/test/internal/layers"
 	"github.com/Microsoft/hcsshim/test/internal/oci"
+	"github.com/Microsoft/hcsshim/test/internal/util"
 	"github.com/Microsoft/hcsshim/test/pkg/require"
-	"github.com/Microsoft/hcsshim/test/pkg/uvm"
+	testuvm "github.com/Microsoft/hcsshim/test/pkg/uvm"
 )
 
 func BenchmarkLCOW_Container(b *testing.B) {
 	requireFeatures(b, featureLCOW, featureContainer)
 	require.Build(b, osversion.RS5)
 
-	ctx := namespacedContext()
-	ls := linuxImageLayers(ctx, b)
+	pCtx := namespacedContext()
+	ls := linuxImageLayers(pCtx, b)
 
-	// Create a new uvm per benchmark in case any left over state lingers
+	// Create a new uVM per benchmark in case any left over state lingers
+
+	// there is (potentially) a memory leak in the Linux GCS that causes "memory usage for cgroup exceeded threshold"
+	// errors for the `/gcs` cgroup to be raised.
+	// so every so often iterations, we re-create the uVM
+	const recreateIters = 100
```

Comment:
Can this issues with the GCS be consistently repro'd? Maybe we should look into fixing that and then get rid of this new UVM creation on every 100 iterations logic?

Reply:
not 100%, but its pretty reliable
I was planning on doing that in a future PR, since it didn't fit here